### DISCUSSION TITLE
Self Hosting Feedback

### DISCUSSION THREAD

**Discussion by prologic** [RocketChat/Rocket.Chat#31075] [General]
Hi Team,

As I cannot ask a basic question or talk to any of your sales or support team without divulging additional personal information about me, I'm posting some feedback here instead. Hope that's okay! -- As an aside, you really do not need to ask for my phone number, country of origin, company name, company size, industry just to ask a basic question. Even more horrifying to me is that you automatically subscribe me and forge me to consent to "additional information":

> By submitting this form you consent to receive more information about Rocket.Chat products, events and updates, according to our [privacy policy](https://docs.rocket.chat/privacy-and-security/privacy-and-security-policies/rocket.chat-privacy-policy). You may unsubscribe at any time.

----

Anyway... Feedback on Self Hosting.

I want to Self Host Rocket.Chat. Fine, after several hours of trying to get a `rocketchat` and `mongodb` container(s) up and running, I finally got something that looked like it was working...

Your own Docker instructions and sample `docker-compose` do not work btw. I tried several times and your set-up there just fails.

Thankfully I was able to utilise @frdmn 's nice work: https://github.com/frdmn/docker-rocketchat which worked for me locally pretty quickly.

So at this point I thought, let's get this up and running in my production cluster with a real-world facing domain, with TLS and the works.

Got the containers up and running behind Traefik, no worries, done this hundreds of times with other services. Went through the wizard and came across the screen saying I had to register to my "Rocket Chat Cloud Account". Odd, I thought, that's not what I was trying to do, but whatever I did it anyway as I didn't really have much of a choice.

Then a few minutes later I got email configuration. But! This didn't come from my instance, oh no, this came from Rocket Chat Cloud?! What the hell?!

Okay at this point I was really confused as to what's going on. I'm trying to self host Rockat.Chat afterall. Anyway I went back to my instance and it let me login with the admin username/password I had setup in teh wizard.

I had earlier read in the docs that I could setup Auth via OpenID (OIDC), but that I had to login with an administrator account and go to Settings and set this up.

Nope 👎

Turns out I can't have this feature without paying for it?! I looked and looked and there just doesn't seem to be a way for me to set-up SSO/OAuth/OpenID without an Enterprise account? Can someone please confirm this?

And this is where my feedback ends. I can't use Rocket Chat the way I want, because you the one feature I really wanted to use doesn't exist in the self hosted instance I set-up. Why? Why do I have to pay for a feature when I'm already paying for the infrastructure, the hardware my time and effort.

It's not that I'm not willing to support the project, or even pay in general, but looking at the pricing page I'm expected to pay a per-seat license for these features, despite self-hosting my own server, on my own infrastructure, on my own hardware, with my own operational time, effort and money.

This is a practise I'm finding quite counter intuitive. I don't require support, but if I did, I'd be happy to pay for it. Why block certain quite useful modern-day features behind "Enterprise". I really don't consider basic SSO/OpenID/OAuth to be "Enterprise" times have changed.

----

**Edit:**

I also wanted to talk a little bout (_which is **very** important btw_) about privacy and security, especially when it comes to Self Hosting. If you intend to properly support Self Hoster(s), you cannot by all rights also be sending off registration to your random cloud service, gathering analytics, and the like, certainly not without consent or alternatives. You might own the product and you might want to ask your users for feedback, by by forcing self-hoster(s) to also automatically consent to sending data upstream to your servers, this is not self hosting at all. This has all kinds of security and privacy concerns I can't even begin to describe.

As it is right now, I have to go figure out how to delete my randomly and automatically created Rocket Chat Cloud account that your software automatically created with no option to opt-out of. I'm not even sure how this has been secured either (Email is not really something you can rely on for security).

This has been a pretty awful experience thus far. 😔

**Comment by markwylde** [comment]
> It's not that I'm not willing to support the project, or even pay in general, but looking at the pricing page I'm expected to pay a per-seat.

I also feel the same. I don't mind paying for a service. I remember with old forum software you would pay $XXX and they would let you run 1 instance with all the features with no limits. Then every 6 months you would renew your license to keep getting updates. Now, with tiered pricing it's become so intrusive and finicky.

**Comment by prologic** [comment]
I finally want to add that I cannot find a way to delete my Rocket Chat Cloud account. This is really disappointing considering I had no choice in the matter, and now you have a bunch of PII, so you're susceptible to a data leak just like any other company that provides a service. Problem is I didn't ask for this. If I intended to pay for something I would have expected to go through a "normal" pay flow, with customer information, some form of payment method, etc. I didn't ask for yet-another-cloud-account to be created for me, I have too many of those already :(

I'd like to have my account deleted please. So I'd appreciate it if someone could find ~5 mins out of their day to contact me (a fellow human) to undo this what I never asked to be done. Thank you! 🙏

**Comment by duven87** [comment]
Don't you have the feeling that every RC update is more buggy?

I'm thinking of switching to Matrix

**Comment by MaHDiaLaGaB** [comment]
This feedback is enough to me to not using this platform and go to Chatwoot and make it more secure

**Comment by fluxoz** [comment]
This platform is hot garbage. the tooling and dev environment set-up are atrocious. I wish them luck with profiting off of the FOSS community and continued maintenance of their spaghetti.

**Comment by dom6770** [comment]
Correct me if I'm wrong, but starting with 6.5.0 it's not possible to self-host RocketChat with more than 25 users?

We (40~ users) used RocketChat now for several months, and now we cannot upgrade any more unless we pay 160 USD A MONTH?! and soon desktop and mobile apps will cease support for versions older than 6 months...

**Reply by david-uhlig** [reply]
You can, just unsubscribe from the Basic plan on /admin/subscription

**Reply by slhck** [reply]
What specifically happens when you do that?

**Reply by david-uhlig** [reply]
@slhck it just reverts to the community edition then.

**Reply by slhck** [reply]
Hmm, the [release notes](https://github.com/RocketChat/Rocket.Chat/releases/tag/6.5.0) make it seem like the registration is still necessary — the "revert to community edition" process seems weird to me when it could just start as a community edition and only with an upgrade make people register their instance. The [website](https://www.rocket.chat/pricing) does not even mention a "community edition".

**Comment by rivendellstuff** [comment]
RocketChat is awful. I ran it for a year then they removed read receipts on newer verisons. So I stopped updating. Then they removed push notifications for anyone not running latest version. Sneaky. So then I just switched to Mattermost. Then RocketChat removed push notifications altogether for non paying people. 

This is their model. Be 'free' heroes and use the self hosted community and its feedback to make a solid solution. Then cut the self hosted community out, one limb at a time to force paid. I dont mind the paid idea but at least grandfather the "beta testers" when you flip it. At this rate they will start charging you per message sent and separately per message received. (laugh now sure, but SMS used to be like that).

RocketChat is going backwards and will die out if they continue this crap. I'll never look at RC again. Yall should move out asap.  Zulip is starting this trend as well.

**Reply by david-uhlig** [reply]
Push notifications are still available for community edition instances. Limited at 10,000 per month, just like before.

**Reply by rivendellstuff** [reply]
reread what I said.

tack on "as long as youre on the current version of rocket chat" to the end of your sentence and then its correct.  Its not correct otherwise.
Current version moves previously free features to paid. see the point?  unless thats changed in the last month or two

EDIT: the community edition is new then?  so you can be on any version and still get push notifications as long as youre under 25 users? if thats true. thats good, but RC has already soured the trust. they'll remove more stuff once a bunch of ppl start using the CE. so ya... i stand by my statements

**Reply by david-uhlig** [reply]
No point in arguing with unfriendly people who are confidently incorrect. Community edition has been around for a while, push notifications are available for everyone regardless of the number of users on the instance.

**Reply by rivendellstuff** [reply]
@david-uhlig 

its unfortunate you find disagreeing viewpoints "unfriendly". I don't understand how I'm incorrect. After moving read receipts to paid which was free for everyone prior to; I believe version 6? (But I dont remember exactly and not worth looking up anymore) this email went out.

![image](https://github.com/RocketChat/Rocket.Chat/assets/128875345/269d0c1c-1128-4ce7-8028-5f2f8006a652)

This essentially forces users to upgrade to keep push notifications, marketplace apps and user clients. However in doing so they are also forced to now pay if they want features they had previously which were free. 

I fail to see where I'm "confidently incorrect". I will agree in that its pointless arguing with someone that is "confidently incorrect" though. So I'll leave it at that. 

@cross-hello 

Not officially. I wrote a script I used to migrate mine. I only had 15ish users and under 20GB of attachments to move at the time so the script worked fine for my use-case. Its in my github if you're interested. I think theres a more robust solution out there (ive seen a few recent reddit threads in self-hosted on it) you could search for. They may be overkill though depending on how large your installation is.

**Reply by david-uhlig** [reply]
I mean everyone can see your edits, I quote:

> damn. the biggest problem with the world is ppl cant read apparently

which is pretty unfriendly in my book.

**Reply by rivendellstuff** [reply]
ah yes. I re-read your post and edited realizing I made a mistake in the distro you were referring to. I edited it within 5mins actually. But lets talk about that.

So right now in current times we have an issue where people like you refuse to read and still decide to spew crap at others despite actually reading the content. People like me will look and see if I made a mistake and fix it. If it was much later I would have re-replied, but I reread what I write to ensure its accurate so I saw the issue within mins and edited it. 

Your response to this is to not challenge any of the facts I presented including an email from RocketChat directly proving my points, but instead its to try and shame someone for attempting to correct something. So if people like you will shame others for actually correcting themselves, why would anyone ever admit they're wrong then? Theres only shaming to be had isnt there? We want people to be better, review facts and change their minds to learn and grow don't we? hmm.....

I take it back, what I said originally now still stands about you: **damn. the biggest problem with the world is ppl cant read apparently**

Thank you for proving my point. :) 
Try and be better in the new year. Its not too late.
